packages/number-input/src/Input/Input.tsx

```diff
@@ -47,21 +54,25 @@ export const Input = React.forwardRef<HTMLInputElement, InputProps>(
     forwardRef,
   ) => {
     /**
-     * Timeout for error icon transitions
+     * Timeout for error icon transitions, using a ref so its not recreated after each render.
      */
-    let translateTimeout: NodeJS.Timeout;
+    const translateTimeout = useRef<NodeJS.Timeout>();
+    const containerRef = useRef<HTMLDivElement | null>(null);
     const [shouldErrorTransition, setShouldErrorTransition] =
       useState<boolean>(false);
+    const isFocusedRef = useRef<boolean>(false);
     const inputRef = useForwardedRef<HTMLInputElement | null>(forwardRef, null);
     const { theme } = useDarkMode();
+    const isControlled = !isUndefined(valueProp);
 
     useEffect(() => {
+      // On unmount, removes the timeout that is set on `onMouseLeave` and `'onBlur` of the wrapper container.
       return () => {
-        clearTimeout(translateTimeout);
+        clearTimeout(translateTimeout.current as unknown as number);
```

Comment:
😷 Why do we need `as unknown as number`?

This should work a bit better:
```ts
if (translateTimeout.current) {
  clearTimeout(translateTimeout.current)
}
```

Reply:
yeah this is better, i'll update